packages/roosterjs-editor-plugins/lib/plugins/TableCellSelection/TableCellSelection.ts

```diff
@@ -188,20 +188,27 @@ export default class TableCellSelection implements EditorPlugin {
                 clonedVTable.selection = this.tableRange;
                 removeCellsOutsideSelection(clonedVTable);
                 clonedVTable.writeBack();
-
                 event.range.selectNode(clonedTable);
 
                 if (event.isCut) {
-                    forEachSelectedCell(this.vTable, cell => {
-                        if (cell?.td) {
-                            deleteNodeContents(cell.td, this.editor);
-                        }
-                    });
+                    this.deleteTableColumns(selection.table, selection.coordinates);
                 }
             }
         }
     }
 
+    private deleteTableColumns(table: HTMLTableElement, selection: TableSelection) {
+        const isWholeColumnSelection = table.rows.length - 1 === selection.lastCell.y;
```

Comment:
do you also need to check selection.firstCell.y? What if I select from a middle row to the last row?